src/main/java/edu/kit/kastel/mcse/ardoco/tlr/rest/api/service/ConfigurationProviderService.java

```diff
@@ -0,0 +1,117 @@
+/* Licensed under MIT 2025. */
+package edu.kit.kastel.mcse.ardoco.tlr.rest.api.service;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
```

Comment:
To check: Do we use log4j in this project? I thought that we use slf4j here as well.